src/basic/Tabs/DefaultTabBar.js

```diff
@@ -5,152 +5,169 @@ import _ from "lodash";
-    const fontWeight = isTabActive ? "bold" : "normal";
-    // const fontSize = tabFontSize;
-    if (typeof name === "string") {
-      return (
-        <Button
-          style={{ flex: 1 }}
-          key={name}
-          onPress={() => onPressHandler(page)}
-        >
-          <TabHeading
-            style={isTabActive ? activeTabStyle : tabStyle}
-            active={isTabActive}
-          >
-            <Text
-              style={[
-                { fontSize: tabFontSize },
-                isTabActive ? activeTextStyle : textStyle
-              ]}
-            >
-              {name}
-            </Text>
-          </TabHeading>
-        </Button>
-      );
-    } else {
-      return (
-        <Button
-          style={{ flex: 1 }}
-          key={_.random(1.2, 5.2)}
-          onPress={() => onPressHandler(page)}
-        >
-          <TabHeading style={tabHeaderStyle} active={isTabActive}>
-            {headerContent}
-          </TabHeading>
-        </Button>
-      );
-    }
-  },
+    renderTab(
+        name,
+        page,
+        isTabActive,
+        onPressHandler,
+        tabStyle,
+        activeTabStyle,
+        textStyle,
+        activeTextStyle,
+        tabHeaderStyle,
+        tabFontSize,
+        disabled
+    ) {
+        const headerContent =
+            typeof name !== "string" ? name.props.children : undefined;
+        const { activeTextColor, inactiveTextColor } = this.props;
+        const textColor = isTabActive ? activeTextColor : inactiveTextColor;
+        const fontWeight = isTabActive ? "bold" : "normal";
+        // const fontSize = tabFontSize;
+        if (typeof name === "string") {
+            return (
+                <Button
+                    style={{ flex: 1 }}
+                    key={name}
+                    onPress={() => {
+                        if (disabled === true) {
+                            return;
+                        }
+                        onPressHandler(page);
+                    }}
+                >
+                    <TabHeading
+                        style={isTabActive ? activeTabStyle : tabStyle}
+                        active={isTabActive}
+                    >
+                        <Text
+                            style={[
+                                { fontSize: tabFontSize },
+                                isTabActive ? activeTextStyle : textStyle
+                            ]}
+                        >
+                            {name}
+                        </Text>
+                    </TabHeading>
+                </Button>
+            );
+        } else {
+            return (
+                <Button
+                    style={{ flex: 1 }}
+                    key={_.random(1.2, 5.2)}
+                    onPress={() => {
+                        if (disabled === true) {
+                            return;
+                        }
+                        onPressHandler(page);
```

Comment:
Would something like this work?
```
onPress={() => disabled ? {} : onPressHandler(page)}
```

Just to make the code a bit more concise.

Reply:
I improved the code and I resolved the conflict thanks for your review.